packages/core/src/html/linkProcessor.js

```diff
@@ -87,18 +87,28 @@ function convertMdAndMbdExtToHtmlExt(node) {
     if (hasNoConvert) {
       return;
     }
+
     const { href } = node.attribs;
-    const hasMdExtension = href.slice(-3) === '.md';
-    if (hasMdExtension) {
-      const newHref = `${href.substring(0, href.length - 3)}.html`;
-      node.attribs.href = newHref;
+
+    const hrefUrl = url.parse(href);
+
+    // get the first instance of URL fragment (first encounter of hash)
+    const urlFragment = hrefUrl.hash === null ? '' : hrefUrl.hash;
+    const urlFragmentIdx = urlFragment === '' ? href.length : href.indexOf(urlFragment);
+
+    let ext = '';
+    ext = href.substring(urlFragmentIdx - 3, urlFragmentIdx) === '.md' ? '.md' : ext;
+    ext = href.substring(urlFragmentIdx - 4, urlFragmentIdx) === '.mbd' ? '.mbd' : ext;
+    if (!ext) {
+      // extension is neither .md nor .mbd, we do not need to process the link
       return;
     }
-    const hasMbdExtension = href.slice(-4) === '.mbd';
-    if (hasMbdExtension) {
-      const newHref = `${href.substring(0, href.length - 4)}.html`;
-      node.attribs.href = newHref;
-    }
+
+    const pathWithoutExtAndFragmentLen = href.length - ext.length - urlFragment.length;
```

Comment:
can be simplified slightly with `hrefUrl.pathname` too

Reply:
Yup, that's more succinct. Thanks for the suggestion!